### ISSUE TITLE
All data table's headers aren't identified programmatically

### ISSUE THREAD

**Issue by mathisscott** [vmware/clarity#1934]
URL: http://clarity-ks.surge.sh/tables

Tools: Web Accessibility Toolbar (WAT2.0) 

Reproduced Steps:
1. Login Clarity Kitchen Sink with web browser.
2. Go to Tables tab.
3. Check all data tables in Tables tab.

Actual Results:
All data table’s headers are not identified programmatically. 
For more details, please refer to "Tables-HeadresAreNotIdentified.png". 

Expected Results:
All data table’s headers should be identified programmatically.

Fix Suggestion:
1. Each column header must have either scope="col-header name" or id="x".
2. Each row header must have either scope="row-header name" or id="x".
3. The headers attribute for each data cell must include all the superior parent's header, in order for the column headers to be properly identified.
For more details, please refer to "Table'sMarkup.png".

Note:
This issue also occurs on Typography tab.

Violation of Section 508 Standards: 
Section 1194.22 (g) Row and column headers shall be identified for data tables.

User Impact: Blind.

Screen shot:
1.	Tables-HeadresAreNotIdentified.png

![image.png](https://images.zenhubusercontent.com/582f4cffb9dc3c6220222f57/44edbd1e-ed93-41a4-8dc5-8e179ccea0be)

![image.png](https://images.zenhubusercontent.com/582f4cffb9dc3c6220222f57/564262ab-0c19-423c-ba83-0698a5df0c2b)

**Comment by Jinnie** [comment]
I don't fully understand the requirement:
`Each column header must have either scope="col-header name" or id="x".
`
According to W3C standards, only applicable values for scope are col, row, colgroup, rowgroup.
https://www.w3.org/TR/1999/REC-html401-19991224/struct/tables.html#adef-scope

Please, provide reference for the use of `scope="col-header name"`

**Comment by mathisscott** [comment]
@Jinnie 
This isn't that big of a deal, unless it requires an update to the datagrid. [Here is an example of use](https://developer.mozilla.org/en-US/docs/Learn/HTML/Tables/Advanced#The_scope_attribute).

If this is something that requires changes to the datagrid code, it is out of the scope of an update to kitchen sink. Please move it out of this epic and back into the backlog if so. Thanks.

**Comment by Jinnie** [comment]
After a research done by me on the topic, we have two options:
1) Add scope attribute to th elements
2) Add ids to th elements, and refer them from headers attributes in td elements.
One or the other, no need to do both.
I tried to reproduce it with Mac's VoiceOver, but it was navigating the table well with or without the extra options. Then I used the Chrome extension ChromeVox, which was also navigating inside the table, although not reporting that a cell belongs to a specific header.

So, I will go for option 1) - to only add scope=[row/col], as it seems cleaner and with less code clutter in the td elements.

**Comment by mathisscott** [comment]
@Jinnie 
1 sounds like a great option. Thanks for looking into this!
